compiler/noirc_frontend/src/monomorphization/printer.rs

```diff
@@ -78,6 +80,14 @@ impl AstPrinter {
             }
             Expression::Break => write!(f, "break"),
             Expression::Continue => write!(f, "continue"),
+            Expression::Clone(expr) => {
+                self.print_expr(expr, f)?;
+                write!(f, ".clone()")
```

Comment:
I don't know if this can happen, but would parentheses be needed around `expr` in any case?

Reply:
I don't think so. Any infix or prefix expressions which may require them should already be parenthesized. If-expressions won't be but `if a { b } else { c }.clone()` is still unambiguous.